src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,162 @@
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    INamedTypeSymbol systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol =compilationStartAnalysisContext.Compilation.GetTypeByMetadataName(
+                            WellKnownTypes.SystemNetSecurityRemoteCertificateValidationCallback);
+                    if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol == null)
+                    {
+                        return;
+                    }
+                    
+                    compilationStartAnalysisContext.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            IDelegateCreationOperation delegateCreationOperation =
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+                            
+                            if (delegateCreationOperation.Type == systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol)
+                            {
+                                bool flag = true;
+
+                                var kind = delegateCreationOperation.Target.Kind;
+                                switch (kind)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        flag = dealWithAnonmousFunction(delegateCreationOperation.Target);
+                                        break;
+
+                                    case OperationKind.MethodReference:
+                                        IMethodReferenceOperation methodReferenceOperation = (IMethodReferenceOperation)delegateCreationOperation.Target;
+                                        flag = dealWithMethodReference(methodReferenceOperation);
+                                        break;
+                                }
+
+                                if (!flag)
+                                    operationAnalysisContext.ReportDiagnostic(
+                                        Diagnostic.Create(
+                                            Rule,
+                                            delegateCreationOperation.Syntax.GetLocation(),
+                                            kind.ToString()));
+                            }
+                        },
+                        OperationKind.DelegateCreation);
+                });
+        }
+
+        private static bool dealWithAnonmousFunction(IOperation operation)
+        {
+            bool flag = false;
+
+            var descendants = operation.Descendants();
+            foreach (var tmp in descendants)
+            {
+                if (tmp.Kind == OperationKind.Return)
+                {
+                    IReturnOperation a=(IReturnOperation)tmp;
+                    bool returnedValue = (bool)a.ReturnedValue.ConstantValue.Value;
```

Comment:
You cannot assume the returned value will always be non-null and you also cannot assume that non-null value will always have a constant value.

Reply:
Have some test cases for something like:
```
interface IBoolGetter
{
    bool Get();
}

class TestClass
{
    public IBoolGetter BoolGetter;

    void TestMethod()
    {
         ServicePointManager.ServerCertificateValidationCallback += (a, b, c, d) => BoolGetter.Get();
    }
}
```